[BOP-1475] Improve custom worker profiles docs (#272)
Mirantis/mke-docs: content/docs/configuration/kubernetes/kubelet.md

            nodefs.inodesFree: 5%
```

{{< callout type="warning" >}}
When creating the custom profile, ensure the following:

- Crosscheck `featureGates` in the custom profile against the official
  Kubernetes [list of removed feature gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates-removed/).
  Adding a removed feature gate will prevent the kubelet from starting. 

> - Add `allowedUnsafeSysctls` as namespaced `sysctls`. Non-namespaced `sysctls`

Reviewer: I am not sure I understand what you mean by "as namespaced". 
I had a typo in my initial description that might have made it unclear.
There are two types of sysctls: namespaced and non-namespaced. They may look identical because this particular setting doesn't imply specifying the namespace, it only needs the name of the sysctl. So, the second sentence here is correct, we do not support non-namespaced, and the first sentence should say that only namespaced ones are supported. Maybe my English isn't good enough, but the phrase "add as namespaced" doesn't read right to me.

I found example of the same idea in OpenShift docs for reference - https://docs.openshift.com/container-platform/4.11/nodes/containers/nodes-containers-sysctls.html#namespaced-and-node-level-sysctls

Author: @byDimasik I changed the problematic sentence, I hope it is clearer now.

            nodefs.inodesFree: 5%
```

{{< callout type="warning" >}}

> When creating the custom profile, ensure the following:

Reviewer 2: The transition between the lead-in and the bullets is jarring. Please reword -- either, or -- to make for a smoother communication.

- Crosscheck `featureGates` in the custom profile against the official
  Kubernetes [list of removed feature gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates-removed/).
  Adding a removed feature gate will prevent the kubelet from starting. 
- When setting `allowedUnsafeSysctls` include only namespaced `sysctls`.
  Non-namespaced `sysctls` are unsupported by the kubelet and will also prevent
  it from starting.
{{< /callout >}}

### Apply a custom profile to a node

> Hosts can be assigned a custom profile through the `hosts` section of the MKE

Reviewer 2: ```suggestion
You can assign a custom profile through the `hosts` section of the MKE
```

configuration file, whereas the profile name is an installation time argument
for the host.

The following example configuration applies the `hardworker` custom profile to
the `localhost` node:

```yaml
hosts:
- role: single
  installFlags:
    - --profile=hardworker
  ssh:
    address: localhost
    keyPath: ~/.ssh/id_rsa
    port: 22
    user: root
```

### Debug worker profiles

When an invalid worker profile is provided, the kubelet assigned to use it may
fail to start. If nodes appear in the `NotReady` state after applying a worker
profile, it is likely due to an incorrect worker profile configuration.

To debug your worker profile:

1. SSH into the corresponding `NotReady` node.
2. Check the logs of k0sworker system service `journalctl -u k0sworker`. 
3. If the worker node does not show any errors, SSH into a manager node.
4. Check the logs of the k0scontroller system service
   `journalctl -u k0scontroller`. 
5. Repeat the process for every manager node until you find kubelet or worker

>    profile-related errors.

Reviewer 2: Unless the purpose of this step is to send the user back to Step 1 of the procedure, I suggest that it be wrapped into the preceding step.

"Check the logs of the k0scontroller system service `journalctl -u k0scontroller`. Repeat the process for every manager node until you find errors related to kubelet or to worker profiles."
